dpsim/src/MNASolverDirect.cpp

```diff
@@ -42,6 +42,16 @@ void MnaSolverDirect<VarType>::switchedMatrixStamp(std::size_t index, std::vecto
 	for (UInt i = 0; i < mSwitches.size(); ++i)
 		mSwitches[i]->mnaApplySwitchSparseSystemMatrixStamp(bit[i], sys, 0);
 
+	/* TODO: FIX!!
+	 * in matrix stamping, there are unnecessary nonzeros introduced. They need to be removed,
+	 * otherwise the linear solver can't solve the system - for unknown reasons for now. Probably because
+	 * the matrix indices are not sorted - at least that's where KLU fails
+	 *
+	 * pruning fixes this, but might not be the most efficient and elegant solution
+	 * (it must be known to developer to prune the matrix after each stamp)
+	 */
```

Comment:
Yes, but then it would be executed at each `*SystemMatrixStamp`, which could be inefficient for many calls (or not, it depends how efficient this method is)

Reply:
Calling of `makeCompressed` is ensured now via calls in `mnaApplySystemMatrixStamp` and `mnaApplySwitchSystemMatrixStamp`. @LennartSchu Add here a comment on your findings regarding performance.

Reply:
added comment

Reply:
Adding `makeCompressed` to `mnaApplySystemMatrixStamp` and `mnaApplySwitchSystemMatrixStamp` does not affect performance in the tested models. `makeCompressed` is "inexpensive" according to Eigen (for reference, [see here](https://gitlab.com/libeigen/eigen.git))